### ISSUE TITLE
Un-linking pod dynamic frameworks from a parent dynamic framework

### ISSUE THREAD

**Issue by bryanboyko** [CocoaPods/CocoaPods#7443]
I would like to create a dynamic framework A that can be _manually_ integrated into MyProj.xcworkspace where MyProj.xcworkspace uses Cocoapods to integrate framework B and framework A also uses Cocoapods to integrate framework B. 

Sorry if this has been answered before, but I did some Google digging and wasn't able to find a solution. 

## What did you do?

Created a dynamic framework A.framework. Used Cocoapods to install dependency B in A.framework with use_frameworks!. Then built A.framework and manually integrated it into MyProj.xcworkspace, which uses Cocoapods to separately include dependency B.

```Project MyProj.xcworkspace ==(Manual)=> A.framework ==(Cocoapods)=> B.framework```

## What did you expect to happen?

Pod B was not linked in A.framework so that MyProj could build without linking duplicate frameworks

## What happened instead?

Pod B was included in A.framework, so MyProj.xcworkspace failed to run with errors like this one

```objc[46401]: Class FNFPlayerLayer is implemented in both /Users/bryanboyko/Library/Developer/CoreSimulator/Devices/304E3757-2898-4B8E-87CE-A23B1A709436/data/Containers/Bundle/Application/5C3EF737-7269-4A55-B080-D66897DAEB8D/ObjCSampleApp.app/Frameworks/MAX.framework/MAX (0x108ccdfa8) and /Users/bryanboyko/Library/Developer/CoreSimulator/Devices/304E3757-2898-4B8E-87CE-A23B1A709436/data/Containers/Bundle/Application/5C3EF737-7269-4A55-B080-D66897DAEB8D/ObjCSampleApp.app/ObjCSampleApp (0x102326740). One of the two will be used. Which one is undefined.```

(Class FNFPlayerLayer comes from B)

### Stack

```
   CocoaPods : 1.4.0
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.6 (16G29)
       Xcode : 9.2 (9C40b)
         Git : git version 2.14.3 (Apple Git-98)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ c76425efeaea05de768e6ef722b86baa058534c9
               maxads - https://github.com/MAXAds/Specs.git @ 58c1c0279879e0375c07ab825bf014b27d1e5f58
               Specs - git@github.com:MAXAds/Specs.git @ b69cc151dafcfb28d6e2648b9de84e6231ef472b
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile (A)

```ruby
platform :ios, '8.0'

source 'https://github.com/CocoaPods/Specs.git'

def common_deps

  use_frameworks!

  pod 'B'
 
end


target 'A' do
 
  common_deps

end
```

### Podfile (MyProj)

```ruby
platform :ios, '8.0'

source 'https://github.com/CocoaPods/Specs.git'

def common_deps

  use_frameworks!

  pod 'B'
 
end


target 'MyProj' do
 
  common_deps

end
```

## Project that demonstrates the issue

Let me know if an example project will be useful and I can create a sample project that demonstrates this

**Comment by dnkoutso** [comment]
By default CocoaPod will link pods to the top level (app) targets.

What you are asking is more or less a re-write of this mechanism which is definitely right but probably be included in a CocoaPods 2.0.0 release.

The only way around this is by building machinery (ready hackery?) around `Podfile` and `post_install` hook to edit the settings of your project.

**Comment by segiddins** [comment]
If you're looking to use CocoaPods to produce binary artifacts, I highly reccomend taking a look at [cocoapods-packager](https://github.com/CocoaPods/cocoapods-packager)

**Comment by segiddins** [comment]
Packager is what I'd reccomend, although it is unmaintained at the moment

**Comment by bryanboyko** [comment]
@segiddins alright thanks for getting back so soon

Will Cocoapods PR https://github.com/CocoaPods/CocoaPods/pull/6966 for Swift static libraries allow app level ObjC files to import Swift files from a Pod?

**Comment by segiddins** [comment]
Yes, although you've always been able to `@import` Swift pods from Objective-c

**Comment by bryanboyko** [comment]
@import is not working when static_framework = true 

Here is an example project and dependency:
- Pod installing in ObjCSampleApp2 root directory and then running ObjCSampleApp2.xcworkspace runs successfully (static_framework = true is commented)
- Then uncommenting static_framework = true SampleSwiftSource.podspec and pod installing again in ObjCSampleApp2 prevents SampleSwiftSource Swift files from being recognized

[ObjCSampleApp2.zip](https://github.com/CocoaPods/CocoaPods/files/1749914/ObjCSampleApp2.zip)
[SampleSwiftSource.zip](https://github.com/CocoaPods/CocoaPods/files/1749915/SampleSwiftSource.zip)

Here is a related issue https://github.com/CocoaPods/CocoaPods/issues/7307

**Comment by segiddins** [comment]
cc @paulb777

**Comment by paulb777** [comment]
Acknowledged - if still open, will investigate the static_framework import issue when I next have CocoaPods bandwidth - probably about 2 weeks.

**Comment by bryanboyko** [comment]
@segiddins @paulb777 I hate to pester, but do you guys have an idea when the next Cocoapods pre-release will be? We have some new SDK integrations coming up and it would be great to get access to static libs on Cocoapods master :)

**Comment by segiddins** [comment]
No clue

**Comment by paulb777** [comment]
The static framework build problem is solved with #7519 with the caveat of needing to add a dummy swift source file to the app for the link to succeed.

**Comment by bryanboyko** [comment]
@paulb777 thanks! This is excellent!

**Comment by dnkoutso** [comment]
@paulb777 should this be now closed that #7519 is now merged?

**Comment by paulb777** [comment]
I'm ok to close.

As noted, there is still an open corner case that may never come up in practice of an Objective C-only  app using a Swift static framework CocoaPod.

@bryanboyko what do you think?

**Comment by bryanboyko** [comment]
@paulb777 to which corner case are you referring? The caveat you mentioned a couple comments above?

```
'the caveat of needing to add a dummy swift source file to the app for the link to succeed.'
```

Or one of the cases from #7307?

My main goal to is to be able to integrate a mixed Swift/ObjC SDK into both Swift and Objective-C. The integration into Objective-C being more important since almost all of our potential customers have apps in Objective-C. 

If, after #7519, an Objective-C app only needs to add a dummy Swift file to consume whatever it wants from a mixed Swift/ObjC static framework, my main goal is met. 

That being said, I opened this ticket because I was trying to use Cocoapods to install dependencies into my SDK and then package the SDK to be manually imported into third party projects. Per @segiddins comment above, it sounds like that would be a job for Cocoapods Packager, which wasn't working a number of weeks ago when I tested it due to CocoaPods/cocoapods-packager#115. 

If this is more of a packager issue, it makes sense to me to close this out. Once CocoaPods/cocoapods-packager#115 is solved then it'll be possible to test the specific objective of this issue. If it's still outstanding, it can be reopened in the Packager repo?

**Comment by dnkoutso** [comment]
@bryanboyko I think he meant the "adding a dummy swift source file"

**Comment by bryanboyko** [comment]
@dnkoutso awesome. Fine to close from my end then. 

Packager issue is still outstanding, but much lower priority for me if a normal Cocoapods integration is possible.

**Comment by dnkoutso** [comment]
@bryanboyko thanks, will close this out for now. Please do re-open if something is still missing or not working.

Worth using latest master version of CocoaPods if you want to try the mixed stuff because it just landed yesterday. (#7519)

**Comment by bryanboyko** [comment]
Sounds good. Thanks I'll check it out!
